fabric-rendering-v1/src/client/java/net/fabricmc/fabric/mixin/client/rendering/InGameHudMixin.java

```diff
@@ -16,21 +16,198 @@
+	)
+	private LayeredDrawer wrapOverlayMessage(LayeredDrawer instance, LayeredDrawer.Layer layer) {
+		return wrap(OVERLAY_MESSAGE, instance, layer);
+	}
+
+	@Redirect(method = "<init>", at = @At(
+			value = "net.fabricmc.fabric.impl.client.rendering.LayerInjectionPoint",
+			target = "Lnet/minecraft/client/gui/hud/InGameHud;renderTitleAndSubtitle(Lnet/minecraft/client/gui/DrawContext;Lnet/minecraft/client/render/RenderTickCounter;)V")
+	)
+	private LayeredDrawer wrapTitleAndSubtitle(LayeredDrawer instance, LayeredDrawer.Layer layer) {
+		return wrap(TITLE_AND_SUBTITLE, instance, layer);
+	}
+
+	@Redirect(method = "<init>", at = @At(
+			value = "net.fabricmc.fabric.impl.client.rendering.LayerInjectionPoint",
+			target = "Lnet/minecraft/client/gui/hud/InGameHud;renderChat(Lnet/minecraft/client/gui/DrawContext;Lnet/minecraft/client/render/RenderTickCounter;)V")
+	)
+	private LayeredDrawer wrapChat(LayeredDrawer instance, LayeredDrawer.Layer layer) {
+		return wrap(CHAT, instance, layer);
+	}
+
+	@Redirect(method = "<init>", at = @At(
+			value = "net.fabricmc.fabric.impl.client.rendering.LayerInjectionPoint",
+			target = "Lnet/minecraft/client/gui/hud/InGameHud;renderPlayerList(Lnet/minecraft/client/gui/DrawContext;Lnet/minecraft/client/render/RenderTickCounter;)V")
+	)
+	private LayeredDrawer wrapPlayerList(LayeredDrawer instance, LayeredDrawer.Layer layer) {
+		return wrap(PLAYER_LIST, instance, layer);
+	}
+
+	@Redirect(method = "<init>", at = @At(
+			value = "net.fabricmc.fabric.impl.client.rendering.LayerInjectionPoint",
+			target = "Lnet/minecraft/client/gui/hud/InGameHud;method_55806(Lnet/minecraft/client/gui/DrawContext;Lnet/minecraft/client/render/RenderTickCounter;)V")
+	)
+	private LayeredDrawer wrapSubtitlesHud(LayeredDrawer instance, LayeredDrawer.Layer layer) {
+		return wrap(SUBTITLES, instance, layer);
+	}
+
+	@Redirect(method = "<init>",
+			at = @At(
+				value = "net.fabricmc.fabric.impl.client.rendering.LayerInjectionPoint",
+				target = "Lnet/minecraft/client/gui/hud/InGameHud;renderSleepOverlay(Lnet/minecraft/client/gui/DrawContext;Lnet/minecraft/client/render/RenderTickCounter;)V"
+			)
+	)
+	private LayeredDrawer wrapSleepOverlay(LayeredDrawer instance, LayeredDrawer.Layer layer) {
+		return wrap(SLEEP, instance, layer);
+	}
+
+	@Unique
+	private static LayeredDrawer wrap(Identifier identifier, LayeredDrawer instance, LayeredDrawer.Layer layer) {
+		return instance.addLayer(IdentifiedLayer.wrapping(identifier, layer));
+	}
+
+	@Redirect(method = "<init>", at = @At(value = "INVOKE", target = "Lnet/minecraft/client/gui/LayeredDrawer;addSubDrawer(Lnet/minecraft/client/gui/LayeredDrawer;Ljava/util/function/BooleanSupplier;)Lnet/minecraft/client/gui/LayeredDrawer;", ordinal = 0))
+	private LayeredDrawer wrapSubDrawer1(LayeredDrawer instance, LayeredDrawer drawer, BooleanSupplier shouldRender) {
```

Comment:
Would it be better to handle the sub drawers by mixing into addSubDrawer? They wouldn't even need IDs that way?

Reply:
There could be some problems with flattening all sub drawers, but I'll try that.